code/controllers/subsystem/statpanel.dm

```diff
@@ -50,63 +50,114 @@ SUBSYSTEM_DEF(statpanels)
 	while(length(currentrun))
 		var/client/target = currentrun[length(currentrun)]
 		currentrun.len--
-		var/ping_str = url_encode("Ping: [round(target.lastping, 1)]ms (Average: [round(target.avgping, 1)]ms)")
-		var/other_str = url_encode(json_encode(target.mob.get_status_tab_items()))
-		target << output("[encoded_global_data];[ping_str];[other_str]", "statbrowser:update")
+		if(!target.statbrowser_ready)
+			target << browse(file('html/statbrowser.html'), "window=statbrowser")
+			continue
```

Comment:
I fail to see how repeatedly reloading the  statbrowser more then *twice a second* is suppose to help.

I can guarantee it will hurt.

It can take byond upwords of 60 seconds to load the IE webview after getting the html file, you just need to have it wait, don't try to reload, have it wait, and add some system for the user to request a reload.

Reply:
like, you needed the statbrowser_ready check, don't get me wrong.

that was the part you needed.

Reply:
Using a callback that checks if the browser is ready after the initial browse call in client now, if that fails it sends the client a message that refreshes the browser when clicked